main/src/main/java/cgeo/geocaching/connector/al/ALApi.java

```diff
@@ -319,90 +347,93 @@ private static List<Geocache> importCachesFromJSON(final Response response) {
             cache.setArchived(response.get("IsArchived").asBoolean());
-            cache.setHidden(parseDate(response.get("PublishedUtc").asText()));
+
             if (!Settings.isALCfoundStateManual()) {
-                cache.setFound(response.get("IsComplete").asBoolean());
+                final boolean isComplete = response.get("IsComplete").asBoolean();
+                cache.setFound(isComplete);
             }
-            final Geocache oldCache = DataStore.loadCache(geocode, LoadFlags.LOAD_CACHE_OR_DB);
-            final String personalNote = (oldCache != null && oldCache.getPersonalNote() != null) ? oldCache.getPersonalNote() : "";
-            cache.setPersonalNote(personalNote, false);
+
             DataStore.saveCache(cache, EnumSet.of(SaveFlag.CACHE));
             return cache;
+
         } catch (final NullPointerException e) {
             Log.e("_AL ALApi.parseCache", e);
             return null;
         }
     }
 
-    // Having a separate parser for details is required because the API provider
-    // decided to use different upper/lower case wordings for the same entities
-
     @Nullable
     private static Geocache parseCacheDetail(final JsonNode response) {
+        final String geocode = ALConnector.GEOCODE_PREFIX + response.get("Id").asText();
         try {
-            final Geocache cache = new Geocache();
-            final JsonNode location = response.at(LOCATION);
-            final String firebaseDynamicLink = response.get("FirebaseDynamicLink").asText();
-            final String[] segments = firebaseDynamicLink.split("/");
-            final String geocode = ALConnector.GEOCODE_PREFIX + response.get("Id").asText();
+            Geocache cache = DataStore.loadCache(geocode, LoadFlags.LOAD_CACHE_OR_DB);
+            if (cache == null) {
+                cache = new Geocache();
+            }
+            populateCommonCacheFields(cache, response);
+
             final String ilink = response.get("KeyImageUrl").asText();
             final String desc = response.get("Description").asText();
-            cache.setGeocode(geocode);
-            cache.setCacheId(segments[segments.length - 1]);
-            cache.setName(response.get(TITLE).asText());
             cache.setDescription((StringUtils.isNotBlank(ilink) ? "<img src=\"" + ilink + "\"></img><p><p>" : "") + desc);
-            cache.setCoords(new Geopoint(location.get(LATITUDE).asText(), location.get(LONGITUDE).asText()));
-            cache.setType(ADVLAB);
-            cache.setSize(CacheSize.getById("virtual"));
-            cache.setVotes(response.get("RatingsTotalCount").asInt());
-            cache.setRating(response.get("RatingsAverage").floatValue());
-            // cache.setArchived(response.get("IsArchived").asBoolean()); as soon as we're using active mode
-            // cache.setFound(response.get("IsComplete").asBoolean()); as soon as we're using active mode
-            cache.setDisabled(false);
```

Comment:
Did you leave out `setDisabled()` and user note handling (see old lines 394ff) intentionally?

Reply:
`personalNote` handling is still there, it  happens in:

`Geocache cache = DataStore.loadCache(geocode, LoadFlags.LOAD_CACHE_OR_DB);`

(Lines 368 and 393)
It is an emulated property. If we don't have  that cache in the DatasStore. we don't  have a `personalNote`. There is nothing to parse from the API response. If we have it in the DataStore, `cache` will have `personalNote` already set. Simplifying this logic was a key point in the refactoring, also the introduction of `populateCommonCacheFields()`.

Now the `setDisabled()`, I need to check what happened it is perhaps safest not merging that PR in the meantime as we definitely to not want to end up with any regression in function.

Thx much for pointing me to it!

Reply:
IIRC this extra personal note handling got introduced some time ago as personal note was overwritten when updating an ALC. But I may be wrong or that info may be outdated.

Reply:
What version of AndroidStudio are you using currently? Here's a compatibility matrix for AS & AGP: https://developer.android.com/build/releases/gradle-plugin#android_gradle_plugin_and_android_studio_compatibility

I'm using AS 2024.3.2 (aka "Meerkat Feature Drop"), which runs just fine with current settings.

Reply:
@baiti The method Geocache.gatherMissingFrom will do the merging of the online and offline data.
Inside that method the waypoints are tried to "merged" via Geocache.mergeWaypoints.

But the waypoints themself shouldn't have a "online" user note, only the cache has a user note. So maybe (and I am not sure right now how that works in this scenario) the parsing of the caches user note may overwrite the waypoints user note (as you can store the waypoint-information in the user note and they are kind of automatically restored from there)